Assets/MRTK/Providers/Oculus/XRSDK/OculusXRSDKDeviceManager.cs

```diff
@@ -0,0 +1,74 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using Microsoft.MixedReality.Toolkit.Input;
+using Microsoft.MixedReality.Toolkit.Utilities;
+using Microsoft.MixedReality.Toolkit.XRSDK.Input;
+using System;
+using UnityEngine.XR;
+
+namespace Microsoft.MixedReality.Toolkit.XRSDK.Oculus
+{
+    /// <summary>
+    /// Manages XR SDK devices on the Oculus platform.
+    /// </summary>
+    [MixedRealityDataProvider(
+        typeof(IMixedRealityInputSystem),
+        SupportedPlatforms.WindowsStandalone | SupportedPlatforms.Android,
+        "XRSDK Oculus Device Manager")]
+    public class OculusXRSDKDeviceManager : XRSDKDeviceManager
+    {
+        /// <summary>
+        /// Constructor.
+        /// </summary>
+        /// <param name="inputSystem">The <see cref="Microsoft.MixedReality.Toolkit.Input.IMixedRealityInputSystem"/> instance that receives data from this provider.</param>
+        /// <param name="name">Friendly name of the service.</param>
+        /// <param name="priority">Service priority. Used to determine order of instantiation.</param>
+        /// <param name="profile">The service's configuration profile.</param>
+        public OculusXRSDKDeviceManager(
+            IMixedRealityInputSystem inputSystem,
+            string name = null,
+            uint priority = DefaultPriority,
+            BaseMixedRealityProfile profile = null) : base(inputSystem, name, priority, profile) { }
+
+        #region Controller Utilities
+
+        /// <inheritdoc />
+        protected override Type GetControllerType(SupportedControllerType supportedControllerType)
+        {
+            switch (supportedControllerType)
+            {
+                case SupportedControllerType.OculusTouch:
+                    return typeof(OculusXRSDKTouchController);
+                default:
+                    return base.GetControllerType(supportedControllerType);
+            }
+        }
+
+        /// <inheritdoc />
+        protected override InputSourceType GetInputSourceType(SupportedControllerType supportedControllerType)
+        {
+            switch (supportedControllerType)
+            {
+                case SupportedControllerType.OculusTouch:
+                    return InputSourceType.Controller;
+                default:
+                    return base.GetInputSourceType(supportedControllerType);
+            }
+        }
+
+        /// <inheritdoc />
+        protected override SupportedControllerType GetCurrentControllerType(InputDevice inputDevice)
+        {
+            if (inputDevice.characteristics.HasFlag(InputDeviceCharacteristics.Controller))
```

Comment:
Nit: Curious about consistency - why does this function use an if () but the other ones use switch () (where they are roughly doing the same thing?)

Reply:
Characteristics is an unity.xr-side inputdevice property which is separate from our own SupportedControllerType definition. Characteristics is a bitmask that defines several properties that a input device can take on (controll, handtracking, headmounted, etc)